xml/registry.rnc

```diff
@@ -136,10 +137,12 @@ Commands = element commands {
 #     opcode - numeric opcode of specified type for this function
 #     name - if present, protocol name (defaults to command name)
 #     comment - unused
+ArgumentClass = attribute class { "buffer" | "display list" | "framebuffer" | "program" | "program pipeline" | "query" | "renderbuffer" | "sampler" | "sampler" | "shader" | "sync" | "texture" | "transform feedback" | "vertex array" }
```

Comment:
I don't think these should be defined in the rnc spec, as frankly it'll just be a pain when updating the spec later down the line. One of the big focuses of #335 was making it easy for spec editors to use awesome features like groups - if a spec editor had to update the rnc spec when writing a new extension then they're much less likely to use the attribute at all.

Reply:
But while groups are added by most extensions - there is pretty limited number of classes, so it's hard to apply the same logic.
More strict checks would help extensions that don't add new classes, making it harder only to make mistakes, while giving pretty descriptive error log:
```
gl.xml:11767:37: error: value of attribute "class" is invalid; must be equal to "buffer", "display list", "framebuffer", "program", "program pipeline", "query", "renderbuffer", "sampler", "shader", "sync", "texture", "transform feedback" or "vertex array"
```

On other hand, extensions that would add new classes are rare enough, it can be fixed by either me or anyone else personally interested.
And such extensions tend to be pretty complex, so i don't think adding few strings to the list in `registry.rnc` would be much of a leap for their author.

Reply:
There are a lot more classes than the ones in this PR today. Personally would prefer classes to be treated the same as groups in the schema - if a lot of work is required for spec authors to add new class attributes then they just won't do it at all. This was the main focus of my group respecification so that adding groups is less of a barrier for spec authors.

Reply:
```suggestion
ArgumentClass = attribute class { text }
```

Reply:
It won't affect most generators, as at least how I'm planning to use these attributes is create new strongly-typed classes for every new attribute value we come across in the spec.

Reply:
> if a lot of work is required for spec authors to add new class attributes then they just won't do it at all

I just explained why i think it generally wouldn't make spec authors work harder, please don't just ignore.

> Most generators don't even look into the rnc spec.

I don't think `.rnc` should be used for generators ever, it's only for tests in this repository.
And white it wouldn't be reasonable to test group names using `.rnc` (even if it wouldn't be pain to then add new groups) - IMO it makes all the sense for classes, as i explained above.

Reply:
The RNC spec should very rarely ever have to change. Before the latest group changes, the RNC spec hadn't been updated for **3 years** (aside from copyright headers and other meta changes), and before that it was also pretty stagnant since the XML spec's inception in 2013 (**7 years ago**). However, this change means that whenever a new object is introduced in an extension, the RNC spec must also be updated.

I think there should be a clear division between data and schema, as a schema is meant to outline a "structure" and nothing more. Schemas become a lot less portable and more than just a structural guideline when tied down with specifics.

Reply:
> I think there should be a clear division between data and schema

Well i guess you are right in some way... There should be a better way to have this kind of check.